src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -770,20 +768,33 @@ private static ReadOnlySpan<char> EatAllWhitespace(ReadOnlySpan<char> str)
             str[i] == '0' &&
             (str[i + 1] | 0x20) == 'x';
 
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        private void WriteByteHelper(Span<byte> destination)
+        // Returns an unsigned byte array containing the GUID.
+        public byte[] ToByteArray()
         {
-            if (Sse2.IsSupported)
+            var g = new byte[16];
+            if (BitConverter.IsLittleEndian)
             {
-                fixed (byte* dstPtr = &destination[0])
-                fixed (int* thisPtr = &_a)
-                {
-                    Sse2.Store((int*) dstPtr, Sse2.LoadVector128(thisPtr));
-                }
+                MemoryMarshal.TryWrite<Guid>(g, ref this);
+            }
+            else
+            {
+                TryWriteBytes(g);
+            }
+            return g;
+        }
+
+        // Returns whether bytes are sucessfully written to given span.
+        public bool TryWriteBytes(Span<byte> destination)
+        {
+            if ((uint)destination.Length < 16)
```

Comment:
Is this bounds check duplicated in `MemoryMarshal.Write` ?

Reply:
@jkotas I guess MemoryMarshal.Write has it's own bounds checks (according to output) and this uint-hack doesn't help.

Reply:
So why not to use TryWrite for the little endian path like I have suggested above?

Reply:
@jkotas oh, I didn't notice it, thanks!